### ISSUE TITLE
Crash: A pixel buffer cannot be appended when readyForMoreMediaData is NO

### ISSUE THREAD

**Issue by Francescu** [svtek/SceneKitVideoRecorder#22]
Issue environment:

- Device Model: iPhone 6S
- Xcode Version: 9
- iOS Version: 11

Issue Details:

Sometimes the recorder crashes, `line 343`

```swift
self?.pixelBufferAdaptor.append(pixelBuffer, withPresentationTime: appendTime)
```
Error message:
```
'NSInternalInconsistencyException', reason: '*** -[AVAssetWriterInputPixelBufferAdaptor appendPixelBuffer:withPresentationTime:] A pixel buffer cannot be appended when readyForMoreMediaData is NO.'
```

**Comment by Francescu** [comment]
I confirm that the crash still occurs often. I dug a little bit to try to understand.

**Apple References**

> When expectsMediaDataInRealTime is true, isReadyForMoreMediaData will become false only when the input cannot process media samples as quickly as they are being provided by the client. If isReadyForMoreMediaData becomes false for a real-time source, the client may need to drop samples or consider reducing the data rate of appended samples.

It seems that it is already that you meant to do. The check is occurring in `updateDisplayLink`. 
```swift
      guard let input = self?.videoInput, input.isReadyForMoreMediaData else { return }
```

So the problem occurs later, I found that appending might be delayed

```swift
SceneKitVideoRecorder.bufferAppendSemaphore.wait()

      bufferQueue.async { [weak self] in
        if self?.videoFramesWritten == false {
          self?.videoFramesWritten = true
          self?.firstVideoTimestamp = currentTime
        }

        self?.pixelBufferAdaptor.append(pixelBuffer, withPresentationTime: appendTime)
        SceneKitVideoRecorder.bufferAppendSemaphore.signal()
      }
```

The check `isReadyForMoreMediaData` may need to be performed again, within the `bufferQueue` closure?

**Comment by okaris** [comment]
This should ne solved in 1.4.0. Can tou make sure you are experiencing this with 1.4.0

**Comment by Francescu** [comment]
Ok let me update, and I come back to you 👌

**Comment by hablema** [comment]
Had the same issue in 1.3.9 using setupmicrophone method. But was solved in the 1.4.0 update.
